.github/workflows/api-baseline-generation.yml

```diff
@@ -0,0 +1,194 @@
+name: API Baseline Generation
+
+on:
+    push:
+        branches:
+            - main
+    workflow_dispatch:
+        inputs:
+            source_branch:
+                description: "Branch to generate baseline from"
+                required: true
+                default: "main"
+                type: string
+            target_repository:
+                description: "Target repository for baseline storage (owner/repo)"
+                required: false
+                default: ""
+                type: string
+            force_regeneration:
+                description: "Force baseline regeneration even without breaking change"
+                required: false
+                default: false
+                type: boolean
+
+env:
+    CARGO_TERM_COLOR: always
+    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    GITHUB_REPOSITORY: ${{ github.repository }}
+
+jobs:
+    manage-baselines:
+        runs-on: ubuntu-latest
+        env:
+            CARGO_TARGET_DIR: ${{ github.workspace }}/target
+
+        steps:
+            - name: Checkout code
+              uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.event.inputs.source_branch || 'main' }}
+                  fetch-depth: 0 # Fetch full history for PR detection
```

Comment:
Sorry, one more thing. Maybe we only need a few commits, instead of the entire history? Perhaps the last 10 commits?